hazelcast/src/test/java/com/hazelcast/map/impl/querycache/QueryCacheBenchmark.java

```diff
@@ -0,0 +1,183 @@
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.results.format.ResultFormatType;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@BenchmarkMode({Mode.AverageTime})
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Warmup(iterations = 2, time = 5, timeUnit = SECONDS)
+@Measurement(iterations = 2, time = 5, timeUnit = SECONDS)
+@Fork(value = 2, jvmArgsAppend = {"-Xms8G", "-Xmx8G"})
+@State(Scope.Benchmark)
+public class QueryCacheBenchmark {
+
+    @Benchmark
+    public Object testValues(BenchmarkContext context) {
+        return context.getQueryCache().values(Predicates.alwaysTrue());
+    }
+
+    @Benchmark
+    public Object testValuesWithPredicate(BenchmarkContext context) {
+        return context.getQueryCache().values(context.getSql());
+    }
+
+    @Benchmark
+    public Object testKeys(BenchmarkContext context) {
+        return context.getQueryCache().keySet(Predicates.alwaysTrue());
+    }
+
+    @Benchmark
+    public Object testKeysWithPredicate(BenchmarkContext context) {
+        return context.getQueryCache().keySet(context.getSql());
+    }
+
+    @Benchmark
+    public Object testEntries(BenchmarkContext context) {
+        return context.getQueryCache().entrySet(Predicates.alwaysTrue());
+    }
+
+    @Benchmark
+    public Object testEntriesWithPredicate(BenchmarkContext context) {
+        return context.getQueryCache().entrySet(context.getSql());
+    }
+
+    @Benchmark
+    public Object testGet(BenchmarkContext context) {
+        return context.getQueryCache().get(context.getRandomKey());
+    }
+
+    @State(Scope.Benchmark)
+    public static class BenchmarkContext {
+
+        private static final String MAP_NAME = "test";
+
+        @Param("100000")
+        protected int entryCount;
+
+        @Param({"1024"})
+        protected int keySize;
+
+        private HazelcastInstance instance;
+        private Random random;
+        private QueryCache queryCache;
+        private SqlPredicate sql = new SqlPredicate("age > 100 AND age < 3000");
+        Object[] keys;
+
+        @Setup(Level.Trial)
+        public void setUp() {
+            Config config = new Config();
+            MapConfig mapConfig = new MapConfig(MAP_NAME);
+            QueryCacheConfig queryCacheConfig = new QueryCacheConfig(MAP_NAME);
+            queryCacheConfig.setInMemoryFormat(InMemoryFormat.OBJECT);
+            queryCacheConfig.setSerializeKeys(false);
```

Comment:
How this worked in `5.0.z` benchmark? Did you comment out this line? Did you do any other changes like this while testing on the `5.0.z`?

Reply:
I believe i only commented out this line: `queryCacheConfig.setSerializeKeys(false);`